src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -1826,23 +1934,33 @@ private static bool ParametersSatisfyFunction(IEdmOperation operation, IList<str
                 parametersToMatch = parametersToMatch.Skip(1);
             }
 
-            List<IEdmOperationParameter> functionParameters = parametersToMatch.ToList();
-
             // if any required parameters are missing, don't consider it a match.
-            if (functionParameters.Where(
-                p => !(p is IEdmOptionalParameter)).Any(
-                    p => parameterNameList.All(
-                        k => !string.Equals(k, p.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))))
+            BitArray bitmask = new BitArray(parameterNameList.Count);
+            foreach (var functionParameter in parametersToMatch)
             {
-                return false;
+                bool matched = false;
+                for (int j = 0; j < parameterNameList.Count; j++)
+                {
+                    if (string.Equals(parameterNameList[j], functionParameter.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))

```

Comment:
>parameterNameList[j] [](start = 38, length = 20)

same here for the [j] on IList<T>, it's not random access. What's the benefit to access using "index"?  Maybe I miss something.

Reply:
random access is faster. 

---
In reply to: [363456913](https://github.com/OData/odata.net/pull/1602#discussion_r363456913) [](ancestors = 363456913)